server/src/spacon/components/mqtt/core.clj

```diff
@@ -21,10 +21,31 @@
             [clojure.core.async :as async]
             [spacon.components.http.auth :refer [get-token]]
             [clojure.tools.logging :as log]
-            [spacon.components.queue.protocol :as queue])
+            [spacon.components.queue.protocol :as queue]
+            [clojure.data.json :as json])
   (:import (org.eclipse.paho.client.mqttv3 MqttException)
            (java.net InetAddress)))
 
+(def vcap_mqtt
+  "Map of keys and values from VCAP_SERVICES mqtt environment variable"
+  (or (some-> (System/getenv "VCAP_SERVICES")
+              (json/read-str :key-fn clojure.core/keyword) :p-rabbitmq first :credentials :protocols :mqtt)
+                {:username :username
+                 :password :password
+                 :host :host
+                 :port :port
+                 :ssl :ssl}))
+
+(def mqtt-connect-ops
+  "Map of authentication keys and values for mqtt connect options"
+  {:username (or (System/getenv "MQTT_USERNAME") (:username vcap_mqtt))
+   :password (or (System/getenv "MQTT_PASSWORD") (:password vcap_mqtt))})
+
+(def vcap_tcp
+  "When vcap_mqtt exists it returns a formated tcp connection"
+  (when (some? vcap_mqtt)
```

Comment:
Seems like are trying to define the url to use so this should always return a string....so I would remove this line b/c vcap_mqtt should always return the defaults that you can use to build the mqtt url.